Add Slack support and make it work

benoittgt opened the pull request:
I'm not a huge fan of recursion in the slack module, especially how I manage errors and status. Need a refactoring. 
I move on the top the method with the empty list to properly pattern matching... That's ugly.

Also need to find a way to do proper testing with external api and VCR. The token has been regenerated for security...

Commit:
Add slack support to forward github msg to slack

I'm not a huge fan of recursion in slack module, especially how I manage errors and status. Need a refactoring. Also need to find a way to do proper testing with external api and VCR. The token has been regenerated for security...

nicoolas25 commented on lib/phubme/slack.ex:
```diff
@@ -0,0 +1,25 @@
+defmodule PhubMe.Slack do
+  def send_private_message({:ok, _comment, [], _sender, _source}) do
```
Since, the first element of the tuple is always `:ok`, you may not need it. What do you think?

benoittgt replied:
yep I agree

Commit:
Keep only matched nicknames in NicknamesMatcher

Commit:
Add slack into Plug and remove :ok in NicknamesMatcher

Also had to disable `async` in Plug test because it was causing error.
Don't know why.

Commit:
Remove unused "parsed_comment"

benoittgt commented:
Ok. I did a test with a payload with matching nickname on my slack and good slack bot token and it's work.

`curl -vX POST http://localhost:4000/ -d @test/fixtures/github_issue_comment_request.json --header "Content-Type: application/json"`

And slackbot : 
<img width="923" alt="capture d ecran 2016-09-14 a 23 31 22" src="https://cloud.githubusercontent.com/assets/8417720/18530780/617378d6-7ad3-11e6-831e-6a537b967c9c.png">

Still lot's of thing to change but as a 0.1 seems good.
